engine/src/test/java/org/hibernate/validator/test/internal/engine/ValidatorTest.java

```diff
@@ -211,6 +214,57 @@ public void testValidateValueWithNestedPath() {
 		Set<ConstraintViolation<X>> constraintViolations = validator.validateValue( X.class, "list[0].foo", null );
 		assertNumberOfViolations( constraintViolations, 1 );
 		assertCorrectPropertyPaths( constraintViolations, "list[0].foo" );
+
+		Set<ConstraintViolation<K>> constraintViolationsK = validator.validateValue( K.class, "foo.bar", null );
+		assertNumberOfViolations( constraintViolationsK, 1 );
+		assertCorrectPropertyPaths( constraintViolationsK, "foo.bar" );
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HV-1002")
+	public void testValidatePropertyWithNestedPath() {
+		Validator validator = getValidator();
+		X someX = new X();
+		someX.addZ( new Z() );
+		Set<ConstraintViolation<X>> constraintViolationsX = validator.validateProperty( someX, "list[0].foo" );
+		assertNumberOfViolations( constraintViolationsX, 1 );
+		assertCorrectPropertyPaths( constraintViolationsX, "list[0].foo" );
+
+		I someI = new I();
+		someI.putJ( "bar", new J() );
+		Set<ConstraintViolation<I>> constraintViolationsI = validator.validateProperty( someI, "map[bar].foo" );
+		assertNumberOfViolations( constraintViolationsI, 1 );
+		assertCorrectPropertyPaths( constraintViolationsI, "map[bar].foo" );
+
+		K someK = new K();
+		someK.foo = new L<String>();
+		Set<ConstraintViolation<K>> constraintViolationsK = validator.validateProperty( someK, "foo.bar" );
+		assertNumberOfViolations( constraintViolationsK, 1 );
+		assertCorrectPropertyPaths( constraintViolationsK, "foo.bar" );
+
+		constraintViolationsK = validator.validateProperty( someK, "foo.genericProperty" );
+		assertNumberOfViolations( constraintViolationsK, 1 );
+		assertCorrectPropertyPaths( constraintViolationsK, "foo.genericProperty" );
+	}
+
+	@Test(expectedExceptions = { ValidationException.class }, expectedExceptionsMessageRegExp = "^HV000195:.*")
+	@TestForIssue(jiraKey = "HV-1002")
+	public void testValidatePropertyWithNestedPathAndNullPropertyInTheWay() {
+		Validator validator = getValidator();
+		X someX = new X();
+		Set<ConstraintViolation<X>> constraintViolationsX = validator.validateProperty( someX, "list[0].foo" );
+		assertNumberOfViolations( constraintViolationsX, 1 );
```

Comment:
Seems these assertions are never reached due to the (expected) exception before?